test/extensions/transport_sockets/alts/tsi_frame_protector_test.cc

```diff
@@ -0,0 +1,102 @@
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/transport_sockets/alts/tsi_frame_protector.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "src/core/tsi/fake_transport_security.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Alts {
+
+using testing::InSequence;
+using testing::Invoke;
+using testing::NiceMock;
+using testing::SaveArg;
+using testing::Test;
+using testing::_;
+using namespace std::string_literals;
+
+/**
+ * Test with fake frame protector. The protected frame header is 4 byte length (little endian,
+ * include header itself) and following the body.
+ */
+class TsiFrameProtectorTest : public Test {
+public:
+  TsiFrameProtectorTest() : frame_protector_({tsi_create_fake_frame_protector(nullptr)}) {}
+
+protected:
+  TsiFrameProtector frame_protector_;
+};
+
+TEST_F(TsiFrameProtectorTest, Protect) {
+  {
+    Buffer::OwnedImpl input, encrypted;
+    input.add("foo");
+
+    EXPECT_EQ(TSI_OK, frame_protector_.protect(input, encrypted));
+    EXPECT_EQ("\x07\0\0\0foo"s, encrypted.toString());
```

Comment:
nits: comment where the prefix "\x07\0\0\0" comes from? I guess fake_frame_protector has a contract about how to generate it.

Reply:
There is comment at top of this file (L23)